### PR TITLE
Samrooke/ac 2008 workflow manager delete payload

### PR DESCRIPTION
**Pull request by samrooke**
### Description

Fixes # .

A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] All tests passed locally.
- [ ] [Documentation comments](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/) included/updated.
- [ ] [User guide updated](../docs).
- [ ] I have updated the [changelog](../docs/changelog.md)

### CODE REVIEW

File: src/WorkflowManager/Common/Interfaces/IPayloadService.cs [hunk 1]
```diff
@@ -41,6 +41,12 @@ public interface IPayloadService : IPaginatedApi<Payload>
                                          string? patientId = "",
                                          string? patientName = "");
 
+        /// <summary>
+        /// Deletes a payload by id.
+        /// </summary>
+        /// <param name="payloadId">payload id to delete.</param>
+        Task<bool> DeletePayloadFromMinIOAsync(string payloadId);
```
Comment (woodheadio) [inline src/WorkflowManager/Common/Interfaces/IPayloadService.cs:48]: Please could you remove 'FromMinIO'. The storage layer is theoretically pluggable, so might not be MinIO.

File: tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs [hunk 2]
```diff
@@ -203,5 +225,45 @@ public async Task GetAll_ReturnsPayloads()
 
             result.Should().BeEquivalentTo(payload);
         }
+
+        [Fact]
+        public async Task DeletePayloadFromMinIOAsync_ReturnsTrue()
```
Comment (woodheadio) [inline tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs:230]: Please remove 'FromMinIO'

File: tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs [hunk 3]
```diff
@@ -203,5 +225,45 @@ public async Task GetAll_ReturnsPayloads()
...
+
+            _storageService.Setup(s => s.RemoveObjectsAsync(It.IsAny<string>(), It.IsAny<IEnumerable<string>>(), default));
+
+            var result = await PayloadService.DeletePayloadFromMinIOAsync(payloadId);
+
+            result.Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task DeletePayloadFromMinIOAsync_ThrowsMonaiNotFoundException()
```
Comment (woodheadio) [inline tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs:245]: Please remove 'FromMinIO'

File: tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs [hunk 4]
```diff
@@ -203,5 +225,45 @@ public async Task GetAll_ReturnsPayloads()
...
+
+#pragma warning disable CS8603 // Possible null reference return.
+            _payloadRepository.Setup(p => p.GetByIdAsync(It.IsAny<string>())).ReturnsAsync(() => null);
+#pragma warning restore CS8603 // Possible null reference return.
+
+            await Assert.ThrowsAsync<MonaiNotFoundException>(async () => await PayloadService.DeletePayloadFromMinIOAsync(payloadId));
+        }
+
+        [Fact]
+        public async Task DeletePayloadFromMinIOAsync_ThrowsMonaiBadRequestException()
```
Comment (woodheadio) [inline tests/UnitTests/Common.Tests/Services/PayloadServiceTests.cs:257]: Please remove 'FromMinIO'

**Review by woodheadio — APPROVED** [review]

**Review by neildsouth — APPROVED** [review]

**Pull request merged**